code/modules/unit_tests/mapping.dm

```diff
@@ -17,3 +17,61 @@
 			continue
 
 		TEST_FAIL(log_entry)
+
+/// Verifies that there are no space turfs inside a station area, or on any planetary z-level. Sometimes, these are introduced during the load of the map and are not present in the DMM itself.
+/// Let's just make sure that we have a stop-gap measure in place to catch these if they pop up so we don't put it onto production servers should something errant come up.
+/datum/unit_test/mapload_space_verification
+
+/datum/unit_test/mapload_space_verification/Run()
+	// Grab information on our current station.
+	var/datum/map_config/current_map = SSmapping.config
+	var/current_space_ruin_levels = current_map.space_ruin_levels
+	var/current_empty_space_levels = current_map.space_empty_levels
+
+	// Is our current map a planetary station (NO space turfs allowed)? If so, check for ANY space turfs.
+	if(!(current_empty_space_levels && current_space_ruin_levels))
+		validate_planetary_map()
+		return
+
+	// Let's explicitly outline valid areas to catch anything that we're... okay with for now, because there are some valid use cases (though we should try to avoid them).
+	var/list/excluded_area_typecache = typecacheof(list(
+		// Space! This is likely an intentional space turf, so let's not worry about it.
+		/area/space,
+		// Transit areas have space turfs in a valid placement.
+		/area/shuttle/transit,
+		// Space Ruins do their own thing (dilapidated lattices over space turfs, for instance). Rather than fuss over it, let's just let it through.
+		/area/ruin/space,
+		// Same stipulation as space ruins, but they're (ruined) shuttles instead.
+		/area/shuttle/ruin,
+		// Solars have lattices over space turfs, and are a valid placement for space turfs in a station area.
+		/area/station/solars,
+	))
+
+	// We aren't planetary, so let's check area placements and ensure stuff lines up.
+	for(var/turf/iterated_turf in world)
```

Comment:
This is really slow and so it should be in its own priority after all the faster tests.

@LemonInTheDark needs to do the same thing with one of his PRs. Work together to come up with a name and value so that you don't conflict with each other when I merge on or the other.

Reply:
Priority assigned

Reply:
Should be worth mentioning that according to runners this test takes roughly 0.6 seconds, while stuff like `mob_faction` (3.5 seconds), `preferences_implement_everything` (5.4 seconds), and `screenshot_humanoids` (31.3 seconds). I have no issue with giving it a priority if you need, but I'm just wondering where we want to assign all of the "funny define numbers" if these (and some more that I'm missing) take a magnitude longer to run.

The values are just from one test case, https://github.com/tgstation/tgstation/actions/runs/3367020406/jobs/5584114352, but those values seem accurate cross-comparing to similar CI runs.

Reply:
I just rename the monkey number and used it
Mine is iterating all turfs in the world tho, so it should be a bit more then 0.6 seconds